lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,36 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def respond_to?(name, include_private = false)
+        Injector.instance_methods.include?(name) || builder.respond_to?(name)
```

Comment:
Yes, I did it this way because there's no `#methods` or `#public_methods` for `BasicObject`. And there's also no `#class` on `BasicObject`, so I can't use `self.class.instance_methods`. So this means I must fall back to `Injector.instance_methods`.

This class isn't intended to be subclassed, so I'm not really concerned about that behaviour. Can you think of any reason I should be?

Again, because this is a class with a very specific purpose (offering a way for users to get at the various injector strategies, and _nothing else_) , I also decided not to bother with handling the different visibility levels, because we don't actually have any private methods we care about exposing to the outside.

All of this is because this is a BasicObject and because we're trying to define as narrow as possible an API in a low-method-pollution environment. The alternative is not even to bother implementing a `#respond_to`, since there's nothing compelling us to, and it's seeming from our discussions that it could be a locus of confusion for people working on this code 😉 

Otherwise, we could just leave it like this and be satisfied that it more or less does what we want – tell outside users whether we respond to `:[]` or the various available strategy names 😄